components/aws/sagemaker/test/integration/utils/kfp_client_utils.py

```diff
@@ -0,0 +1,42 @@
+import os
+import json
+import utils
+
+
+def compile_and_run_pipeline(client, experiment_id, test_file_dir, pipeline_name):
+
+    env_value = os.environ.copy()
+    env_value["PYTHONPATH"] = f"{os.getcwd()}:" + os.environ.get("PYTHONPATH", "")
+    pipeline_path = os.path.join(test_file_dir, pipeline_name)
+    utils.run_command(
+        f"dsl-compile --py {pipeline_path}.py --output {pipeline_path}.yaml",
+        env=env_value,
+    )
+    run = client.run_pipeline(experiment_id, pipeline_name, f"{pipeline_path}.yaml")
+    return run.id
+
+
+def wait_for_job_completion(client, run_id, timeout):
+    response = client.wait_for_run_completion(run_id, timeout)
+    status = response.run.status.lower() == "succeeded"
+    return status
+
+
+def compile_run_monitor_pipeline(
+    client, experiment_id, test_file_dir, pipeline_name, timeout
+):
+    run_id = compile_and_run_pipeline(
+        client, experiment_id, test_file_dir, pipeline_name
+    )
+    status = wait_for_job_completion(client, run_id, timeout)
+    return run_id, status
+
+
+def get_workflow_json(client, run_id):
```

Comment:
why can't we just use `.//sdk/python/build/kfp-0.3.0/kfp/_client.py` it has same function.

Reply:
I guess the comment is confusing as to why it refers to code. This method is missing in [API documentation](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html#kfp.Client) and hence I added it here so that we can look at method's input and output if needed

To answer your question as to why we cant use the source - because we are installing a specific version of kfp SDK in the environment and can access the API directly from the SDK.